[Test Failure] integration.states.test_pip_state.PipStateTest

Test failures from https://github.com/saltstack/salt/pull/57238

`TypeError: string indices must be integers`
and
`pip._vendor.distlib.DistlibException: Unable to locate finder for 'pip._vendor.distlib'`

This looks like the following is causing the issue...
```
Traceback (most recent call last):
[2020-05-13T14:22:53.223Z]          File "C:\\Users\\ADMINI~1\\AppData\\Local\\Temp\\kitchen\\testing\tests\support\mixins.py", line 678, in assertSaltTrueReturn
[2020-05-13T14:22:53.223Z]            self.assertTrue(saltret)
[2020-05-13T14:22:53.223Z]          File "c:\python37\lib\unittest\case.py", line 705, in assertTrue
[2020-05-13T14:22:53.223Z]            raise self.failureException(msg)
[2020-05-13T14:22:53.223Z]        AssertionError: False is not true
[2020-05-13T14:22:53.223Z]        
[2020-05-13T14:22:53.223Z]        During handling of the above exception, another exception occurred:
[2020-05-13T14:22:53.223Z]        
[2020-05-13T14:22:53.223Z]        Traceback (most recent call last):
[2020-05-13T14:22:53.223Z]          File "c:\users\admini~1\appdata\local\temp\kitchen\testing\tests\integration\states\test_pip_state.py", line 464, in test_issue_6833_pip_upgrade_pip
[2020-05-13T14:22:53.223Z]            self.assertSaltTrueReturn(ret)
[2020-05-13T14:22:53.223Z]          File "C:\\Users\\ADMINI~1\\AppData\\Local\\Temp\\kitchen\\testing\tests\support\mixins.py", line 684, in assertSaltTrueReturn
[2020-05-13T14:22:53.223Z]            **(next(six.itervalues(ret)))
[2020-05-13T14:22:53.223Z]        AssertionError: False is not True. Salt Comment:
[2020-05-13T14:22:53.223Z]        Failed to install packages: pip==8.0.1. Error: Collecting pip==8.0.1
[2020-05-13T14:22:53.223Z]          Downloading https://files.pythonhosted.org/packages/45/9c/6f9a24917c860873e2ce7bd95b8f79897524353df51d5d920cd6b6c1ec33/pip-8.0.1-py2.py3-none-any.whl (1.2MB)
[2020-05-13T14:22:53.223Z]        Installing collected packages: pip
[2020-05-13T14:22:53.223Z]          Found existing installation: pip 8.0.0
[2020-05-13T14:22:53.223Z]            Uninstalling pip-8.0.0:
[2020-05-13T14:22:53.223Z]              Successfully uninstalled pip-8.0.0
[2020-05-13T14:22:53.223Z]          Rolling back uninstall of pip C:\Users\ADMINI~1\AppData\Local\Temp\salt-tests-tmpdir\6833-pip-upgrade-pip\lib\site-packages\pip\pep425tags.py:89: RuntimeWarning: Config variable 'Py_DEBUG' is unset, Python ABI tag may be incorrect
[2020-05-13T14:22:53.525Z]          warn=(impl == 'cp')):
[2020-05-13T14:22:53.525Z]        C:\Users\ADMINI~1\AppData\Local\Temp\salt-tests-tmpdir\6833-pip-upgrade-pip\lib\site-packages\pip\pep425tags.py:93: RuntimeWarning: Config variable 'WITH_PYMALLOC' is unset, Python ABI tag may be incorrect
[2020-05-13T14:22:53.525Z]          warn=(impl == 'cp')):
```

Virtualenv is trying to install pip 8.0.1 but these issues appear to be resolved in pip 9.0.1+

--- a/tests/integration/states/test_pip_state.py
+++ b/tests/integration/states/test_pip_state.py
@@ -431,7 +431,7 @@ class PipStateTest(ModuleCase, SaltReturnAssertsMixin):
         # Let's install a fixed version pip over whatever pip was
         # previously installed
         ret = self.run_function(
-            "pip.install", ["pip==8.0"], upgrade=True, bin_env=venv_dir
+            "pip.install", ["pip==9.0.1"], upgrade=True, bin_env=venv_dir
         )
 
         if not isinstance(ret, dict):
@@ -444,14 +444,14 @@ class PipStateTest(ModuleCase, SaltReturnAssertsMixin):
         self.assertEqual(ret["retcode"], 0)
         self.assertIn("Successfully installed pip", ret["stdout"])
 
-        # Let's make sure we have pip 8.0 installed
+        # Let's make sure we have pip 9.0.1 installed
         self.assertEqual(
-            self.run_function("pip.list", ["pip"], bin_env=venv_dir), {"pip": "8.0.0"}
+            self.run_function("pip.list", ["pip"], bin_env=venv_dir), {"pip": "9.0.1"}
         )
 
         # Now the actual pip upgrade pip test
         ret = self.run_state(
-            "pip.installed", name="pip==8.0.1", upgrade=True, bin_env=venv_dir
+            "pip.installed", name="pip==20.0.1", upgrade=True, bin_env=venv_dir
         )
 
         if not isinstance(ret, dict):
@@ -462,7 +462,7 @@ class PipStateTest(ModuleCase, SaltReturnAssertsMixin):
             )
 
         self.assertSaltTrueReturn(ret)
-        self.assertSaltStateChangesEqual(ret, {"pip==8.0.1": "Installed"})
+        self.assertSaltStateChangesEqual(ret, {"pip==20.0.1": "Installed"})
 
     @slowTest
     def test_pip_installed_specific_env(self):

